### PR TITLE
Return path inside recursive trace function

### PR DESCRIPTION
**Pull request by zluipaiva**
Related to: #1 

The `__trace_key` function was updated to return the found paths instead of printing.
Now it uses an array of paths that gets updated every time the key is found.

I'm storing each path as an array of dictionary keys so we could do something like: https://stackoverflow.com/a/14692747/16586258

### CODE REVIEW

File: src/trace_dkey/__init__.py [hunk 1]
```diff
@@ -9,29 +10,45 @@ def __process_tuple(tuple_elts: List) -> Tuple:
         ans.append(elt.value)
     return tuple(ans)
 
+def __trace_key(dict_key: Any, dict_value: Any, key, path = []) -> None:
```
Comment (Agent-Hellboy) [inline src/trace_dkey/__init__.py]: type annotate path and key, key can be Any

**Review by Agent-Hellboy — CHANGES_REQUESTED** [review]

**Comment by Agent-Hellboy** [comment]
please change README as well to reflect these changes, and add a section named general info why we are returning list of keys

> I'm storing each path as an array of dictionary keys so we could do something like: https://stackoverflow.com/a/14692747/16586258

**Comment by Agent-Hellboy** [comment]
Thanks @zluipaiva everything looks good, just update the README

**Review by Agent-Hellboy — APPROVED** [review]

**Comment by zluipaiva** [comment]
@Agent-Hellboy done! just realized a small bug, I will open in a new issue because it's already in production

**Comment by Agent-Hellboy** [comment]
I see, thanks man

**Pull request merged**